src/Download.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Storj\Uplink;
+
+use DateTimeImmutable;
+use Exception;
+use FFI;
+use FFI\CData;
+use Generator;
+use IteratorAggregate;
+use Psr\Http\Message\StreamInterface;
+use Storj\Uplink\Exception\IOException;
+use Storj\Uplink\Exception\UplinkException;
+use Storj\Uplink\Internal\Scope;
+use Storj\Uplink\Internal\Util;
+use Storj\Uplink\Psr\ReadStream;
+
+/**
+ * Handle to a remote object
+ *
+ * TODO: can the download be read only once? Should we protect against reading it twice?
```

Comment:
Download streams are not designed to be read twice. How is it possible to read it twice with the current API?

Reply:
The programmer could do

```
$download = $project->downloadObject(...);
$content = $download->readAll();
$content = $download->readAll();
```

The second time it will be empty. Might cause some bugs. I know I do this, forgetting to rewind streams/iterators.